auslib/util/swagger.py

```diff
@@ -0,0 +1,119 @@
+import yaml
+import auslib
+
+from os import path
+
+AUS_DIR = path.dirname(auslib.__file__)
+
+
+class SpecBuilder():
```

Comment:
This class is really great overall, but I think it would be good if you kept all Balrog-specific knowledge out of it. Concretely, this means that consumers should be passing full filepaths (or already read files) to it, and the version should be passed explicitly instead of reference to a version file.